platform/security/handler/security-handler-oidc/src/test/java/org/codice/ddf/security/handler/oidc/OidcHandlerConfigurationImplTest.java

```diff
@@ -0,0 +1,93 @@
+  private static Map<String, Object> invalidProperties;
+
+  private OidcHandlerConfigurationImpl handlerConfiguration;
+
+  @BeforeClass
+  public static void setupClass() {
+    emptyProperties = new HashMap<>();
+
+    validProperties = new HashMap<>();
+    validProperties.put(OidcHandlerConfigurationImpl.IDP_TYPE, "generic");
+    validProperties.put(OidcHandlerConfigurationImpl.CLIENT_ID, "generic-client");
+    validProperties.put(OidcHandlerConfigurationImpl.REALM, "master");
+    validProperties.put(OidcHandlerConfigurationImpl.SECRET, "changeit");
+    validProperties.put(OidcHandlerConfigurationImpl.DISCOVERY_URI, "https://discovery/uri");
+    validProperties.put(OidcHandlerConfigurationImpl.BASE_URI, "https://base/uri");
+    validProperties.put(OidcHandlerConfigurationImpl.SCOPE, "openid profile email");
+    validProperties.put(OidcHandlerConfigurationImpl.USE_NONCE, false);
+    validProperties.put(OidcHandlerConfigurationImpl.RESPONSE_TYPE, "code");
+    validProperties.put(OidcHandlerConfigurationImpl.RESPONSE_MODE, "form_post");
+    validProperties.put(OidcHandlerConfigurationImpl.LOGOUT_URI, "https://logout/uri");
+
+    invalidProperties = new HashMap<>();
+    invalidProperties.put(OidcHandlerConfigurationImpl.IDP_TYPE, "invalid idpType");
+    invalidProperties.put(OidcHandlerConfigurationImpl.CLIENT_ID, "invalid clientId");
+    invalidProperties.put(OidcHandlerConfigurationImpl.REALM, "invalid realm");
+    invalidProperties.put(OidcHandlerConfigurationImpl.SECRET, "invalid secret");
+    invalidProperties.put(OidcHandlerConfigurationImpl.DISCOVERY_URI, "invalid discoveryUri");
+    invalidProperties.put(OidcHandlerConfigurationImpl.BASE_URI, "invalid baseUri");
+    invalidProperties.put(OidcHandlerConfigurationImpl.SCOPE, "invalid scope");
+    invalidProperties.put(OidcHandlerConfigurationImpl.USE_NONCE, "invalid useNonce");
+    invalidProperties.put(OidcHandlerConfigurationImpl.RESPONSE_TYPE, "invalid responseType");
+    invalidProperties.put(OidcHandlerConfigurationImpl.RESPONSE_MODE, "invalid responseMode");
+    invalidProperties.put(OidcHandlerConfigurationImpl.LOGOUT_URI, "invalid logoutUri");
+  }
+
+  @Test
+  public void constructWithNull() {
+    handlerConfiguration = new OidcHandlerConfigurationImpl(null);
+
+    assertThat(handlerConfiguration.isInitialized(), is(false));
+  }
+
+  @Test
+  public void constructWithEmptyProperties() {
+    handlerConfiguration = new OidcHandlerConfigurationImpl(emptyProperties);
+
+    assertThat(handlerConfiguration.isInitialized(), is(false));
+  }
+
+  @Test(expected = ClassCastException.class)
+  public void constructWithInvalidProperties() {
+    handlerConfiguration = new OidcHandlerConfigurationImpl(invalidProperties);
+
+    assertThat(handlerConfiguration.isInitialized(), is(false));
+  }
+
+  /* currently does not initialize due to a backend http call to the DiscoveryUri failing */
+  @Test
+  public void constructWithValidProperties() {
```

Comment:
❗️ Given valid properties, the handler config initialization _should_ succeed, right? If so, let's just ignore this test rather than test for incorrect behavior.